### PR TITLE
Log routes

### PR DESCRIPTION
**Pull request by LoganWright**
If you have:

```swift
let drop = try Droplet(arguments: ["vapor", "routes"])
drop.get("foo") { _ in return "" }
drop.put("foo/bar/:id") { _ in return "" }
```

You get:

```
+------+--------+--------------+
| Host | Method | Path         | 
+------+--------+--------------+
| *    | GET    | /foo         |
|      | PUT    | /foo/bar/:id |
+------+--------+--------------+
```

Also can use from cli

```
vapor run routes
```
or
```
.build/debug/App routes
```

### CODE REVIEW

File: Sources/Vapor/Routing/RoutesLog.swift [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+public class RouteList: Command {
+    public let help: [String] = [
+        "Logs the routes of your application"
+    ]
+
+    public let id: String = "routes"
+    public var console: ConsoleProtocol { return drop.console }
+    public unowned let drop: Droplet
+
+    public required init(_ drop: Droplet) {
```
Comment (tanner0101) [inline Sources/Vapor/Routing/RoutesLog.swift]: Can we just pass the router and the logger? Less coupled that way.

Comment (LoganWright) [inline Sources/Vapor/Routing/RoutesLog.swift]: Yup!

**Comment by vzsg** [comment]
Good stuff!
If this log was available programmatically, not just as a command, it could form the basis of a self-describing interface (e.g. Swagger).

**Comment by LoganWright** [comment]
@vzsg what do you mean, not sure I follow

File: Sources/Vapor/Routing/RoutesLog.swift (hunk 1 above)
Comment (LoganWright) [inline Sources/Vapor/Routing/RoutesLog.swift]: Just commenting here per irl discussion. Passing droplet so if user sets `drop.console` after initializing we use their latest console 👍

**Comment by vzsg** [comment]
If I'm understanding correctly, this PR adds a new command to the droplet: "routes", which prints the registered routes in a pretty table, to the standard output. This is very nice if someone cannot read source code, but also paves way for something more serious.

`--8<--8<--8<--  idea starts here --8<--8<--8<--`

Having a list of all registered routes allows the framework to expose this list in well-defined formats, for example [Swagger](http://swagger.io/specification/), on which many tools build (including their own Swagger UI for API introspection and testing).

So in a theoretical future, the Vapor framework could supply a controller that returns the same information in JSON/YAML, not just as a console command.

`-->8-->8-->8--    end of idea   -->8-->8-->8--`

**Comment by Alex293** [comment]
+1 szvg that would be nice while in dev mode to be able to add a resource/discover route to a resource to get all its routes and parameters !

**Pull request merged**